tests/unit/regression/lars_unittest.cc

```diff
@@ -233,6 +257,41 @@ TEST(LeastAngleRegression, lars_N_LessThan_D)
 	SG_UNREF(labels);
 }
 
+TEST(LeastAngleRegression, lars_N_LessThan_D_32_feature_test)
+{
+	SGMatrix<float32_t> data(5,3);
+	SGVector<float64_t> lab(3);
+	generate_data_n_less_d(data,lab);
+
+	CDenseFeatures<float32_t>* features=new CDenseFeatures<float32_t>(data);
+	SG_REF(features);
+	CRegressionLabels* labels=new CRegressionLabels(lab);
+	SG_REF(labels);
+	CLeastAngleRegression* lars=new CLeastAngleRegression(false);
+	lars->set_labels(labels);
+	lars->train_machine<float32_t>(features);
```

Comment:
great that this works now.
Now let's clean up the interface a bit in the sense that we dont have templated train methods.